pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -539,3 +548,24 @@ func (exp *flowExporter) sendDataSet() (int, error) {
 	klog.V(4).Infof("Data set sent successfully. Bytes sent: %d", sentBytes)
 	return sentBytes, nil
 }
+
+func (exp *flowExporter) findFlowType(record flowexporter.FlowRecord) uint8 {
+	if exp.nodeRouteController == nil {
+		klog.Warningf("Can't find flowType without nodeRouteController")
+		return 0
+	}
+	if exp.nodeRouteController.IPInPodSubnets(record.Conn.TupleOrig.SourceAddress) {
```

Comment:
This will not work for network-policy only mode traffic as the installed node cache indexer is not maintained at all. 
https://github.com/vmware-tanzu/antrea/blob/main/pkg/agent/controller/noderoute/node_route_controller.go#L278

I think we need to check the network subnet of local `nodeAddr` in `nodeConfig` to determine if the IP address is inside cluster or not. @jianjuns Could you please confirm if this behavior is correct or not?

Reply:
Thanks Srikar for pointing out it, added a TODO right now and wait for Jianjun's opinion.

Reply:
Thanks for adding these [lines](https://github.com/vmware-tanzu/antrea/pull/2000/files#diff-50f2dd93220c4c832537231ada7856843db446693becade1eb58f6788cc207d2R553-R558) to address this. 

My initial assessment on node network based identification is not correct and was mixed up with something else. There is no straightforward way to detect PodIPs at Antrea agent when operating in network policy only mode.
Therefore, we are supporting Pod-To-External flows for all traffic modes except for the special mode of network policy only mode. There is no regression here.